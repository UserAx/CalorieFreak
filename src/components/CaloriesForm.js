import React from 'react';
import {SingleDatePicker} from 'react-dates';
import moment from 'moment';

export class CaloriesForm extends React.Component {
    constructor(props){
        super(props);
        this.state = {
            foodname: props.calorie ? props.calorie.foodname : '',
            servings: props.calorie ? (props.calorie.servings).toString() : '',
            caloriesAmount: props.calorie ? (props.calorie.caloriesAmount / 100).toString() : '',
            createdAt: props.calorie ? moment(props.calorie.createdAt) : moment(),
            description: props.calorie ? props.calorie.description : '',
            calanderFocus: false,
            error: undefined
        }
    }

    handleFoodNameChange = (e) => {
        const foodname = e.target.value;
        this.setState(() => ({
            foodname 
        }));
    };

    handleCaloriesChange = (e) => {
        const caloriesAmount = e.target.value;
        if(!caloriesAmount || caloriesAmount.match(/^\d{1,}(\.\d{0,2})?$/)){
        this.setState(() => ({caloriesAmount}));
        }
    };

    handleDescriptionChange = (e) => {
        const description = e.target.value;
        this.setState(()=> ({description}));
    };

    handleServingChange = (e) => {
        const servings = e.target.value;
        if(!servings || servings.match(/^\d{1,}?$/))
        this.setState(() => ({servings}));
    };

    handleDateChange = (createdAt) => {
        if(createdAt){
            this.setState(() => ({createdAt}));
        }
    };

    handleFocusChange = ({focused}) => {
        this.setState(() => ({calanderFocus: focused}));
    };

    onSubmit = (e) => {
        e.preventDefault();
        if(!this.state.foodname || !this.state.servings || !this.state.caloriesAmount){
            this.setState(() => ({error: 'Please fill in the foodname, servings and calories properly before submitting.'}));
        }else {
            this.setState(() => ({error: undefined}));
            this.props.onSubmit({
                //This function is passed from AddExpense component and edit editExpense component.
                //And here it will get the necessary values.
                foodname: this.state.foodname,
                servings: parseInt(this.state.servings),
                description: this.state.description,
                caloriesAmount: parseFloat(this.state.caloriesAmount, 10) * 100,
                //the 10 on (this.state.amount, 10), only returns the last two digits. 
                createdAt: this.state.createdAt.valueOf()
            });
        };
    }
    
    render() {
        return (
            <form className="caloriesform__container" onSubmit={this.onSubmit}>
                {this.state.error && <p className="caloriesform__error">{this.state.error}</p>}
                <input className="caloriesform__input" type="text" placeholder="What food you had today?" autoFocus 
                value={this.state.foodname} onChange={this.handleFoodNameChange}/>
                <input className="caloriesform__input" type="text" placeholder="No. of Servings" value={this.state.servings}
                onChange={this.handleServingChange}/>
                <input className="caloriesform__input"
                type="text" placeholder="I am not a psychic. Tell me how much calories you consumed."
                value={this.state.caloriesAmount} onChange={this.handleCaloriesChange}/>
                <SingleDatePicker
                date = {this.state.createdAt}
                onDateChange = {this.handleDateChange}
                focused = {this.state.calanderFocus}
                onFocusChange = {this.handleFocusChange}
                numberOfMonths = {1}
                isOutsideRange = {() => false}
                />
                <textarea className="caloriesform__textarea" placeholder="Extra notes here." 
                value={this.state.description} onChange={this.handleDescriptionChange}></textarea>
                <div className="button--addcalories">
                    <button className="button button--addcalories--extra" >Save</button>
                </div>
            </form>
        );
    }
}
export default CaloriesForm;