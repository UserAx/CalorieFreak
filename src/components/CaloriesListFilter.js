import React from 'react';
import {connect} from 'react-redux';
import {setTextFilter, sortByCalories, sortByDate, setStartDate, setEndDate} from '../actions/filters';
import {DateRangePicker} from 'react-dates';
// import {DateRangePicker} from 'rsuite';

export class CaloriesListFilter extends React.Component {
    state = {
        calenderFocused : null
    }

    handleTextChange = (e) => {
        const text= e.target.value;
        this.props.setTextFilter(text);
    }

    handleSortByChange = (e) => {
        if(e.target.value === 'date'){
            this.props.sortByDate();
        }else if(e.target.value === 'calories'){
            this.props.sortByCalories();
        }
    }

    onDatesChange = ({startDate, endDate}) => {
        this.props.setStartDate(startDate);
        this.props.setEndDate(endDate);
    }

    onFocusChange = (calenderFocused) => {
        this.setState(() => ({calenderFocused})); 
    }

    render() {
        return (
                // <div className="burgermenu">
                //     <input type="checkbox" className="toggler" />
                //     {/* The checkbox so that we can show the menu when we click this checkbox. */}
                //     <div className="hamburger"><div></div></div>
                //     {/* THe hamburger container that will contain the actual three lines. */}
                // </div>
                
            <div className="listfilter">
                    <div className="listfilter__items">
                        <input className="listfilter__input" type="text" value={this.props.filters.text}
                        onChange={this.handleTextChange} placeholder="Filter by Text"/>
                        <select  className="listfilter__select"
                        value={this.props.filters.sortBy}
                        onChange={this.handleSortByChange}
                        >
                            <option value="date">Sort By Date</option>
                            <option value="calories">Sort By Calories</option>
                        </select>
                    </div>
                    <div>
                        <DateRangePicker
                            startDateId="MyDatePickerStartId"
                            endDateId="MyDatePickerEndId"
                            startDate={this.props.filters.startDate}
                            endDate={this.props.filters.endDate}
                            onDatesChange={this.onDatesChange}
                            focusedInput={this.state.calenderFocused}
                            onFocusChange={this.onFocusChange}
                            numberOfMonths={1}
                            isOutsideRange={() => false}
                            showClearDates={true}/>
                    </div>
            </div>
        )
    }
}

const mapDispatchToProps = (dispatch) => {
    return {
        setTextFilter: (text) => dispatch(setTextFilter(text)),
        sortByCalories: () => dispatch(sortByCalories()),
        sortByDate: () => dispatch(sortByDate()),
        setStartDate: (startDate) => dispatch(setStartDate(startDate)),
        setEndDate: (endDate) => dispatch(setEndDate(endDate))
    }
}

const mapStateToProps = (state) => {
    return {
        filters: state.filters
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(CaloriesListFilter);