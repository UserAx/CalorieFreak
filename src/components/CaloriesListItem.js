import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';

const CaloriesListItem = ({ id, foodname, caloriesAmount, createdAt }) => (
    <Link className = "calorieslistitemlink" to={`edit/${id}`}>
        <div>
            <h2 className = "calorieslistitem__title">{foodname}</h2> 
            <h5 className = "calorieslistitem__date">{moment(createdAt).format('MMMM Do, YYYY')}</h5>
        </div>
        <span className = "calorieslistitem__calories">{caloriesAmount/ 100}</span>
    </Link>
);

export default CaloriesListItem;