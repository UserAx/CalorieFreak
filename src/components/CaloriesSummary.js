import React from 'react';
import {Link} from 'react-router-dom';
import {connect} from 'react-redux';
import caloriesSelector from '../selector/calories';
import caloriesTotal from '../selector/caloriestotal';

export const CaloriesSummary = ({caloriesCount, caloriesTotal}) => (
    <div className="summary__container">
        <div>
            <h3>Viewing <span>{caloriesCount}</span>, a total of {caloriesTotal/100}
            {(caloriesTotal>0)?' caloires':' calorie'}.</h3>
        </div>
        <div>
            <Link className="button summary_container__button--link" to='/addcalories'>Add Record</Link>
        </div>
    </div>
);

const mapStateToProps = (state) => {
    const visibleCalories = caloriesSelector(state.calories, state.filters);
    return {
        caloriesCount: visibleCalories.length,
        caloriesTotal: caloriesTotal(visibleCalories)
    }
};

export default connect(mapStateToProps)(CaloriesSummary);
