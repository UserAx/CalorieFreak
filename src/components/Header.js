import React from 'react';
import {Link} from 'react-router-dom';
import {startLogout} from '../actions/auth';
import {connect} from 'react-redux';

export const Header = ({startLogout}) => (
    <div className="header__container">
        <div className="header__container__linkcontainer">
            <Link to='/'><h1>Calorie Freak</h1></Link>
            <h4>Stay Fit Stay Healthy</h4>
        </div>
        <div>
            <button className="button button--header" onClick={startLogout}>Log Out</button>
        </div>
    </div>
);

const mapDispatchToProps = (dispatch) => ({
    startLogout: () => dispatch(startLogout())
});

export default connect(undefined, mapDispatchToProps)(Header);