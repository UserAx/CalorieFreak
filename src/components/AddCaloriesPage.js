import React from 'react';
import { connect } from 'react-redux';
import CaloriesForm from './CaloriesForm';
import {startAddCalorie} from '../actions/calories';

export class AddCaloriesPage extends React.Component {    
    onSubmit = (calorie) => {
        this.props.startAddCalorie(calorie);
        this.props.history.push('/');
    }

    render() {
        return (
            <div>
                <div>
                    <CaloriesForm onSubmit={this.onSubmit} />
                </div>
            </div>
        )
    }
};

const mapDispatchToProps = (dispatch) => {
    return{
        startAddCalorie: (calorie) => {
            dispatch(startAddCalorie(calorie));
        }
    }
}

export default connect(undefined, mapDispatchToProps)(AddCaloriesPage);