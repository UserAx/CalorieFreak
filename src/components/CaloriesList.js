import React from 'react';
import {connect} from 'react-redux';
import CaloriesListItem from './CaloriesListItem';
import selectCalories from '../selector/calories';

export const CaloriesList = (props) => (
    <div className="calorielist__container">
        <div className="calorielist__header">
            <div>Foods</div>
            <div>Calories</div>
        </div>
        <div className="caloriesList__items">
            {props.calories.length > 0 ?
            (
                props.calories.map((calorie) => {
                    return <CaloriesListItem key={calorie.id} {...calorie}/>
                }) 
            ) : (
                <div className="caloriesList__items__empty">
                    <span>No Records</span>
                </div>
            )}
        </div>
    </div>
);

const mapStateToProps = (state) => {
    return {
        calories: selectCalories(state.calories, state.filters)
    }
}

export default connect(mapStateToProps)(CaloriesList);