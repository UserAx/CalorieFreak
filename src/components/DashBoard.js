import React from 'react';
import CaloriesList from './CaloriesList';
import CaloriesListFilter from './CaloriesListFilter';
import CaloriesSummary from './CaloriesSummary';

const DashBoard = () => (
    <div>
        <CaloriesSummary />
        <CaloriesListFilter/>
        <CaloriesList />
    </div>
);

export default DashBoard;