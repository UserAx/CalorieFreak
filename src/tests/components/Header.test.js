import React from 'react';
import {shallow} from 'enzyme';
import {Header} from '../../components/Header';

test('should render header correctly', () => {
    const wrapper = shallow(<Header startLogout = {()=>{ }}/>);
    expect(wrapper).toMatchSnapshot();
});

test('should envoke startLogin-click button of Header', () => {
    const startLogin = jest.fn();
    const wrapper  = shallow(<Header startLogout={startLogin}/>);
    wrapper.find('button').simulate('click');
    expect(startLogin).toHaveBeenCalled();
});