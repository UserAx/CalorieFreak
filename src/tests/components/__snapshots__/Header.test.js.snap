// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render header correctly 1`] = `
<div>
  <div>
    <Link
      to="/"
    >
      <h1>
        CalorieFreak
      </h1>
    </Link>
    <h4>
      Stay fit, Stay Healthy
    </h4>
  </div>
  <div>
    <button
      onClick={[Function]}
    >
      Logout
    </button>
  </div>
</div>
`;
